tslib/src/functions-throttled/school-import/predicates.ts

```diff
@@ -0,0 +1,60 @@
+import * as RA from 'ramda-adjunct'
+import { ISchoolRecord, EstabTypeCode, EstabTypeGroupCode, EstabStatusCode } from './data-access/ISchoolRecord'
+export type LogFunc = (msg: string) => void
+
+export interface ISchoolImportPredicates {
+  isSchoolOpen (school: ISchoolRecord): SchoolPredicateResult
+  isAgeInRange (targetAge: number, school: ISchoolRecord): SchoolPredicateResult
+  isRequiredEstablishmentTypeGroup (school: ISchoolRecord): SchoolPredicateResult
+}
+
+export class SchoolPredicateResult {
+  constructor (isMatch: boolean, message?: string) {
+    this.isMatch = isMatch
+    this.message = message ? message : ''
+  }
+  isMatch: boolean
+  message: string
+}
+
+export class Predicates implements ISchoolImportPredicates {
+  isSchoolOpen (school: ISchoolRecord): SchoolPredicateResult {
+    // we want to load all schools that are open, proposed to open, proposed to close
+    // this is the same as every school that isn't closed
+    if (school.estabStatusCode === EstabStatusCode.Closed) {
+      // 1 - open, 2 - closed, 3 - open proposed to close, 4 - Proposed to open
+      return new SchoolPredicateResult(false, `Excluding school ${school.urn} it is closed - estabStatusCode is [${school.estabStatusCode}]`)
+    }
+    return new SchoolPredicateResult(true)
+  }
+
+  isAgeInRange (targetAge: number, school: ISchoolRecord): SchoolPredicateResult {
+    if (RA.isNilOrEmpty(school.statLowAge) || RA.isNilOrEmpty(school.statHighAge)) {
+      return new SchoolPredicateResult(false, `Excluding school [${school.urn}] due to missing age fields: obj ${JSON.stringify(school)}`)
+    }
+    const low = Number(school.statLowAge)
+    const high = Number(school.statHighAge)
+    if (low <= targetAge && high >= targetAge) {
+      return new SchoolPredicateResult(true)
+    }
+    return new SchoolPredicateResult(false, `Excluding school ${school.urn} as it does not meet age criteria ${school.statLowAge}-${school.statHighAge}`)
+  }
+
+  isRequiredEstablishmentTypeGroup (school: ISchoolRecord): SchoolPredicateResult {
+
+    switch (school.estabTypeGroupCode) {
+      case EstabTypeGroupCode.localAuthorityMaintainedSchool:
+      case EstabTypeGroupCode.academies:
+      case EstabTypeGroupCode.freeSchool:
+        return new SchoolPredicateResult(true)
+      case EstabTypeGroupCode.specialSchool:
+        return new SchoolPredicateResult(school.estabTypeCode === EstabTypeCode.communitySpecialSchool ||
+          school.estabTypeCode === EstabTypeCode.foundationSpecialSchool)
+      case EstabTypeGroupCode.otherTypes:
+        return new SchoolPredicateResult(school.estabTypeCode === EstabTypeCode.serviceChildrensEducation &&
+          school.leaCode !== 704)
```

Comment:
This magic number seems odd.  "704".

Reply:
you wrote the original code 😄 
in the census import PR i turned this into a const, which i intend to do for this also, once the census import is merged in.

its for schools in gibraltar.

Reply:
I'm thinking that this would be modelled better as an exclusion list, but I'll leave it up to you.